logback-core/src/main/java/ch/qos/logback/core/net/ssl/SSLConfigurableSocket.java

```diff
@@ -60,4 +64,24 @@ public void setWantClientAuth(boolean state) {
         delegate.setWantClientAuth(state);
     }
 
+    public void setDontVerifyHostname(boolean dontVerifyHostname) {
+        if (dontVerifyHostname) {
+            return;
+        }
+        SSLParameters sslParameters = delegate.getSSLParameters();
+        try {
+            /*
+             * SSLParameters.setEndpointIdentificationAlgorithm(...)
+             * is only available from Java 7, so we use reflection
+             * to check whether it is available.
+             */
+            Method setEndpointIdentificationAlgorithmMethod = SSLParameters.class.getMethod("setEndpointIdentificationAlgorithm", String.class);
+            setEndpointIdentificationAlgorithmMethod.invoke(sslParameters, "HTTPS");
+            delegate.setSSLParameters(sslParameters);
+        } catch (NoSuchMethodException e) {
+        } catch (IllegalAccessException e) {
+        } catch (IllegalArgumentException e) {
+        } catch (InvocationTargetException e) {
+        }
```

Comment:
empty catch blocks allow disaster to propagate in a perhaps very unsafe way... better to wrap and rethrow as a `RuntimeException`